### PR TITLE
Add trig functions and tests

### PR DESCRIPTION
**Pull request by spcan**
Add all the `stdlib` trigonometric functions to `uom::si::Angle` that may apply inside a new `impl`
Added some sanity checks.
All added functions:
 - `sin`
 - `sinh`
 - `cos`
 - `cosh`
 - `sin_cos`
 - `tan`
 - `tanh`

### CODE REVIEW

**Comment by iliekturtles** [comment]
Awesome, thanks! I hope to review later today and if not I'll have time during the week.

The `no_std` build is failing right now: https://travis-ci.org/iliekturtles/uom/jobs/576333385#L866. Looks like these functions are only available in `std` and not `core` for some reason. Not sure if there is a work-around yet but you may need to feature gate on `std`.

File: src/si/angle.rs [hunk 1]
```diff
@@ -90,3 +142,58 @@ mod tests {
         }
     }
 }
+
+
+#[cfg(test)]
+mod trig {
+    storage_types! {
```
Comment (iliekturtles) [inline src/si/angle.rs]: Add a "types: Float;" line beneath so that only `f32`/`f64` are tested.

File: src/si/angle.rs [hunk 2]
```diff
@@ -90,3 +142,58 @@ mod tests {
...
+        use tests::Test;
+        use num::Zero;
+        use si::quantities::*;
+
+        #[test]
+        fn sanity() {
+            let zero: Angle<V> = Angle::<V>::from(V::zero());
```
Comment (iliekturtles) [inline src/si/angle.rs]: Local setup can be simplified (double check negative zero):

```diff
-            let zero: Angle<V> = Angle::<V>::from(V::zero());
-            let nzero: Angle<V> = Angle::<V>::from(-1.0 * V::zero());
-
-            // PI constant taken from Rust stdlib source
-            let mut pi: Angle<V> = Angle::<V>::from(V::zero());
-            pi.value = 3.141_592_653_589_793_238_462_643_383_279_502_88;
-
-            // PI / 2 constant taken from Rust stdlib source
-            let mut half: Angle<V> = Angle::<V>::from(V::zero());
-            half.value = 1.570_796_326_794_896_619_231_321_691_639_751_44;
+            let zero = Angle::zero();
+            let nzero = -Angle::zero();
+            let pi = Angle::new::<a::radian>(V::from_f64(PI).unwrap());
+            let half = Angle::new::<a::radian>(V::from_f64(FRAC_PI_2).unwrap());
```

File: src/si/angle.rs [hunk 3]
```diff
@@ -90,3 +142,58 @@ mod tests {
...
+            // PI constant taken from Rust stdlib source
+            let mut pi: Angle<V> = Angle::<V>::from(V::zero());
+            pi.value = 3.141_592_653_589_793_238_462_643_383_279_502_88;
+
+            // PI / 2 constant taken from Rust stdlib source
+            let mut half: Angle<V> = Angle::<V>::from(V::zero());
+            half.value = 1.570_796_326_794_896_619_231_321_691_639_751_44;
+
+            Test::assert_approx_eq(&zero.cos(), &1.0);
+            Test::assert_approx_eq(&nzero.cos(),&1.0);
```
Comment (iliekturtles) [inline src/si/angle.rs]: Add a space after the comma.

File: src/si/angle.rs [hunk 4]
```diff
@@ -90,3 +142,58 @@ mod tests {
...
+
+            Test::assert_approx_eq(&pi.cos(), &-1.0);
+            Test::assert_approx_eq(&half.cos(), &0.0);
+
+            Test::assert_approx_eq(&zero.sin(), &0.0);
+            Test::assert_approx_eq(&nzero.sin(), &0.0);
+
+            //Test::assert_approx_eq(&pi.sin(), &0.0);
+            //Test::assert_approx_eq(&half.sin(), &1.0);
```
Comment (iliekturtles) [inline src/si/angle.rs]: I think this line can be uncommented. Test passed locally for me.

**Comment by spcan** [comment]
Tests are failing on `pi` and `half` due to Float inaccuracy, reverting changes.

**Comment by iliekturtles** [comment]
Thanks for the changes. I'll review this week!

**Comment by iliekturtles** [comment]
Sorry, got caught up with the other PRs. I'll be reviewing this week so we can merge and release a new version.

**Comment by iliekturtles** [comment]
Sorry again for the delays! If you can squash your commits and move the tests currently in `src/tests/system.rs` into `src/si/angle.rs` then I will merge.

**Comment by iliekturtles** [comment]
I pushed the changes to one of my branches and ended up getting a couple compile errors. One is because of Cargo backwards compatibility. More specific dependency selection may be able to solve this issue. There is also an error (https://travis-ci.org/iliekturtles/uom/jobs/599779168#L936) finding some methods. I haven't looked at this yet.

**Comment by spcan** [comment]
The cargo compatibility fails seem to be produced by the "edition" tag/feature.

I don't know why the `cos` function is not found, as I feature-gated the implementation behind the "std" and the "si" features.

Moving it into `src/si/angle.rs` I removed the "si" feature from it. On my machine it compiles correctly and all tests pass.

**Comment by iliekturtles** [comment]
Looks like I missed the "std" feature gate in my changes. I'll see about resolving the `Edition` keyword issue this week and then merge these changes!

**Comment by spcan** [comment]
This branch should be able to be merged. All tests pass, except the ones limited by the `edition` keyword in `Cargo.toml`

**Comment by iliekturtles** [comment]
I squashed and merged. Thanks so much for the PR and sorry it took so long to get it merged.

**Pull request closed without merging**